### PR TITLE
Reworks utensils.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Repaths utensils, moves table knives to utensil type.
- Adds chopsticks.
- Reworks how utensils handle food interactions in the backend to benefit from the new edibility system.
- Changes the Chinese food vendor's wares to be slightly less racist.

Depends on #3681.

## Why and what will this PR improve
Better/cleaner utensil code with some skeleton code towards future features.

## Authorship
Myself.

## Changelog
:cl:
tweak: The Chinese food vendor has slightly different wares.
/:cl:

### CODE REVIEW

File: tools/map_migrations/3681_plates.txt
Comment (out-of-phaze) [inline tools/map_migrations/3681_plates.txt:1]: the new migrations in this file should probably be in `3688_utensils.txt` or something similar

File: code/modules/food/utensils/_utensil.dm [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	return loaded_food ? loaded_food.attack_self(user) : ..()
+
+/obj/item/utensil/handle_eaten_by_mob(var/mob/user, var/mob/target)
+	. = loaded_food ? loaded_food.handle_eaten_by_mob(user, target) : ..()
+	if(QDELETED(loaded_food))
+		loaded_food = null
+		update_icon()
+
+/obj/item/utensil/get_edible_material_amount(var/mob/eater)
+	return loaded_food ? loaded_food.reagents?.total_volume : ..()
```
Comment (out-of-phaze) [inline code/modules/food/utensils/_utensil.dm]:
```suggestion
	return loaded_food ? loaded_food.get_edible_material_amount() : ..()
```

File: code/__defines/misc.dm [hunk 2]
```diff
@@ -315,4 +315,10 @@
...
 // Defines for fluorescence (/atom/var/fluorescent)
 #define FLUORESCENT_GLOWS   1	// Glows when under flourescent light
-#define FLUORESCENT_GLOWING 2	// Currently glowing due to flourescent light
\ No newline at end of file
+#define FLUORESCENT_GLOWING 2	// Currently glowing due to flourescent light
+
+// Flags used for utensil-food interaction.
+#define UTENSIL_FLAG_COLLECT  BITFLAG(0)
+#define UTENSIL_FLAG_SCOOP  BITFLAG(1)
+#define UTENSIL_FLAG_SLICE  BITFLAG(2)
```
Comment (out-of-phaze) [inline code/__defines/misc.dm]: is `UTENSIL_FLAG_SLICE` really necessary when we use `has_edge()` for slicing foods? i'm worried that this is going to mean, for example, you can't slice foods with a survival knife or something like that

File: code/__defines/misc.dm [hunk 3]
```diff
@@ -315,4 +315,10 @@
 
 // Defines for fluorescence (/atom/var/fluorescent)
 #define FLUORESCENT_GLOWS   1	// Glows when under flourescent light
-#define FLUORESCENT_GLOWING 2	// Currently glowing due to flourescent light
\ No newline at end of file
+#define FLUORESCENT_GLOWING 2	// Currently glowing due to flourescent light
+
+// Flags used for utensil-food interaction.
+#define UTENSIL_FLAG_COLLECT  BITFLAG(0)
```
Comment (out-of-phaze) [inline code/__defines/misc.dm]: is there an explanation anywhere of the intended difference between "collecting" and "scooping"?

**Review by out-of-phaze — DISMISSED** [review]

File: code/__defines/misc.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/__defines/misc.dm]: The intent is to not be able to use a fork to eat soup, basically. I can document my plans if you want?

File: code/__defines/misc.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/__defines/misc.dm]: I suppose I should unify that with food slicability down the track. I just intended this for like, slicing part of a steak or a cake before using a fork.

File: code/__defines/misc.dm (hunk 3 above)
Comment (out-of-phaze) [reply code/__defines/misc.dm]: That'd be nice. The distinction makes sense, I just wasn't sure which was which. I guess collecting is soup and scooping is mashed potatoes, ice cream, etc.?

Comment (MistakeNot4892) [reply code/__defines/misc.dm]: Collecting was original spearing but someone pointed out it was kinda odd to be spearing stuff on chopsticks. Scooping is spoons.

Comment (MistakeNot4892) [reply code/__defines/misc.dm]: I've added some comments to the defines.

File: code/__defines/misc.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/__defines/misc.dm]: Added edge/sharp to the utensil logic.

Comment (MistakeNot4892) [reply code/__defines/misc.dm]: Sliceable flag also now handles the existing slicing mechanic.

File: code/__defines/misc.dm [hunk 4]
```diff
@@ -314,5 +314,17 @@
...
+#define FLUORESCENT_GLOWS   1
+/// Currently glowing due to flourescent light
+#define FLUORESCENT_GLOWING 2
+
+// Flags used for utensil-food interaction.
+/// Solid or semi-solid food; chopsticks, forks.
+#define UTENSIL_FLAG_COLLECT BITFLAG(0)
+/// Soft, liquid or semi-liquid food; soups, stews, pudding.
+#define UTENSIL_FLAG_SLICE   BITFLAG(2)
+/// Foods that need to be sliced before eating; steak, grapefruit.
```
Comment (out-of-phaze) [inline code/__defines/misc.dm]: you seem to have swapped the comments here

**Review by out-of-phaze — DISMISSED** [review]

File: code/__defines/misc.dm (hunk 4 above)
Comment (MistakeNot4892) [reply code/__defines/misc.dm]: Ah I've messed up the define order when I wqs putting the comments on a new line.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**